### PR TITLE
Add pre-flight checks

### PR DESCRIPTION
**Pull request by krishvoor**
This PR adds the following

* Pre-flight checks
* Clones the hypershift to temp directory
* Clean up kubeadmin configs and other generated files

Signed-off-by: Krishna Harsha Voora <krvoora@redhat.com>

### CODE REVIEW

File: workloads/hyper-scale/common.sh [hunk 1]
```diff
@@ -51,6 +53,19 @@ setup(){
     rosa verify permissions
 }
 
+pre_flight_checks(){
+    echo "Pre flight checks started"
+    export MULTI_AZ=$(rosa describe cluster -c krvoora-router -o json | jq -r [.multi_az] | jq -r .[])
```
Comment (dry923) [inline workloads/hyper-scale/common.sh:58]: you have your cluster name in there still ;-)

Comment (dry923) [reply workloads/hyper-scale/common.sh:58]: and on the next line as well

File: workloads/hyper-scale/common.sh [hunk 2]
```diff
@@ -156,6 +171,11 @@ cleanup(){
     sleep 10
     ROUTE_ID=$(aws route53 list-hosted-zones --output text --query HostedZones | grep $BASEDOMAIN | grep -v terraform | awk '{print$2}' | awk -F/ '{print$3}')
     for id in $ROUTE_ID; do aws route53 delete-hosted-zone --id=$id || true ; done
+    rm -f *-admin-kubeconfig || true
+    rm -f pull-secret || true
+    rm -rf  kube-burner.tar.gz|| true
```
Comment (dry923) [inline workloads/hyper-scale/common.sh:176]: there is an extra space between rm -rf and kube-burner

File: workloads/hyper-scale/common.sh [hunk 3]
```diff
@@ -51,6 +53,19 @@ setup(){
     rosa verify permissions
 }
 
+pre_flight_checks(){
+    echo "Pre flight checks started"
+    export MULTI_AZ=$(rosa describe cluster -c krvoora-router -o json | jq -r [.multi_az] | jq -r .[])
+    export COMPUTE_NODE=$(rosa describe cluster -c krvoora-router -o json | jq -r [.nodes.compute] | jq -r .[])
+
+    if [ "${MULTI_AZ}" == "true" ] && [ "${COMPUTE_NODE}"/3 == 0]; then
```
Comment (dry923) [inline workloads/hyper-scale/common.sh:61]: I think you want % and not / when checking if the compute nodes are divisible by 3

File: workloads/hyper-scale/common.sh [hunk 4]
```diff
@@ -10,21 +10,23 @@ prep(){
...
+        echo "Clean-up Hypershift binaries.."
         sudo rm /usr/local/bin/hypershift || true
-        git clone -q --depth=1 --single-branch --branch ${HYPERSHIFT_CLI_VERSION} ${HYPERSHIFT_CLI_FORK}    
-        pushd hypershift
+        export TEMP_DIR=$(mktemp -d)
+
+        git clone -q --depth=1 --single-branch --branch ${HYPERSHIFT_CLI_VERSION} ${HYPERSHIFT_CLI_FORK} -v $TEMP_DIR/hypershift
+        pushd $TEMP_DIR/hypershift
         make build
         sudo cp bin/hypershift /usr/local/bin
```
Comment (mukrishn) [inline workloads/hyper-scale/common.sh:20]: This will still overwrite the current CLI version in use.

File: workloads/hyper-scale/common.sh (hunk 1 above)
Comment (krishvoor) [reply workloads/hyper-scale/common.sh]: Thanks for the catch, updated

File: workloads/hyper-scale/common.sh (hunk 2 above)
Comment (krishvoor) [reply workloads/hyper-scale/common.sh]: Thanks for the catch, updated

File: workloads/hyper-scale/common.sh (hunk 3 above)
Comment (krishvoor) [reply workloads/hyper-scale/common.sh]: Thanks for the catch, updated

File: workloads/hyper-scale/common.sh (hunk 4 above)
Comment (dry923) [reply workloads/hyper-scale/common.sh:20]: +1

**Review by mukrishn — APPROVED** [review]
LGTM

File: workloads/hyper-scale/common.sh [hunk 5]
```diff
@@ -51,6 +50,20 @@ setup(){
     rosa verify permissions
 }
 
+pre_flight_checks(){
+    echo "Pre flight checks started"
+    export MULTI_AZ=$(rosa describe cluster -c $MGMT_CLUSTER_NAME -o json | jq -r [.multi_az] | jq -r .[])
+    export COMPUTE_NODE=$(rosa describe cluster -c $MGMT_CLUSTER_NAME -o json | jq -r [.nodes.compute] | jq -r .[])
+
+    if [ "${MULTI_AZ}" == "true" ] && [ "${COMPUTE_NODE}"%3 == 0]; then
```
Comment (smalleni) [inline workloads/hyper-scale/common.sh:58]: Why do we need the compute nodes to be a multiple of 3 in this check? If we deploy rosa/osd cluster with multi-az the cli will only let us give a multiple of 3 for worker node count - so I guess this point is moot.

**Comment by smalleni** [comment]
@mukrishn @dry923 LGTM please merge if you are good

**Comment by dry923** [comment]
@krishvoor Looks good. One request. Can we add a line similar to https://github.com/cloud-bulldozer/e2e-benchmarking/blob/65e41ddce5eff9586e6f3d162df351c7e3ce2f20/workloads/hyper-scale/common.sh#L49 for hypershift -version (or some similar command) to confirm that the compiled command works as expected.

**Comment by krishvoor** [comment]
> @krishvoor Looks good. One request. Can we add a line similar to
> 
> https://github.com/cloud-bulldozer/e2e-benchmarking/blob/65e41ddce5eff9586e6f3d162df351c7e3ce2f20/workloads/hyper-scale/common.sh#L49
> 
> for hypershift -version (or some similar command) to confirm that the compiled command works as expected.

@dry923 Done

**Comment by dry923** [comment]
> > @krishvoor Looks good. One request. Can we add a line similar to
> > https://github.com/cloud-bulldozer/e2e-benchmarking/blob/65e41ddce5eff9586e6f3d162df351c7e3ce2f20/workloads/hyper-scale/common.sh#L49
> > 
> > for hypershift -version (or some similar command) to confirm that the compiled command works as expected.
> 
> @dry923 Done

LG! Thanks!

**Review by dry923 — APPROVED** [review]
LGTM!

**Pull request merged**